lib/rules/no-disallowed-lwc-imports.js

```diff
@@ -0,0 +1,99 @@
+    // Common APIs
+    'LightningElement',
+    'api',
+    'track',
+    'wire',
+
+    // From "@lwc/engine-core"
+    'getComponentDef',
+    'getComponentConstructor',
+    'isComponentConstructor',
+    'createContextProvider',
+    'readonly',
+    'register',
+    'setFeatureFlagForTest',
+    'unwrap',
+
+    // From "@lwc/engine-dom"
+    'hydrateComponent',
+    'buildCustomElementConstructor',
+    'createElement',
+
+    // From "@lwc/engine-server"
+    'renderComponent',
+]);
+
+const isLwcImport = (node) => node.source.type === 'Literal' && node.source.value === 'lwc';
+
+module.exports = {
+    meta: {
+        docs: {
+            description: 'restrict unexpected imports from the lwc package',
+            category: 'LWC',
+            url: docUrl('no-disallowed-lwc-imports'),
+        },
+        schema: [],
+    },
+
+    create(context) {
+        return {
+            ImportDeclaration(node) {
+                if (isLwcImport(node)) {
+                    const { specifiers } = node;
+                    if (!specifiers.length) {
+                        // import 'lwc'
+                        context.report({
+                            message: `Invalid import. Bare imports are not allowed on "lwc". Instead, use named imports: "import { LightningElement } from 'lwc'".`,
+                            node,
+                        });
+                    }
+                    for (const specifier of specifiers) {
+                        const { type, imported } = specifier;
+                        if (type === 'ImportNamespaceSpecifier') {
+                            // import * as lwc from 'lwc'
+                            context.report({
+                                message: `Invalid import. Namespace imports are not allowed on "lwc". Instead, use named imports: "import { LightningElement } from 'lwc'".`,
+                                node: specifier,
+                            });
+                        } else if (type === 'ImportDefaultSpecifier') {
+                            // import lwc from 'lwc'
+                            context.report({
+                                message: `Invalid import. "lwc" does not export a default module. Instead, use named imports: "import { LightningElement } from 'lwc'".`,
+                                node: specifier,
+                            });
+                        } else if (
+                            type === 'ImportSpecifier' &&
+                            !LWC_SUPPORTED_APIS.has(imported.name)
+                        ) {
+                            // import { fake } from 'lwc'
+                            context.report({
+                                message: `Invalid import. ${imported.name} is not part of the lwc api.`,
+                                node: specifier,
+                            });
+                        }
+                    }
+                }
+            },
+            ExportAllDeclaration(node) {
```

Comment:
It should also lint for named export declaration.

```js
export { lwc } from 'lwc';
```

Reply:
Good catch. I also added an error for `export {} from 'lwc'`.